src/Data/StringVariants/Util.hs

```diff
@@ -23,3 +28,39 @@ usePositiveNat n a f = case someNatVal n of
     EQI -> f p
     LTI -> f p
     GTI -> a
+
+-- Logic ported from base-4.17 Data.Char.isSpace.
+-- Hopefully, GHC would add this type-level function
+type IsCharWhitespace c
+  = If
+    (CharToNat c <=? 0x376)
+    (CharToNat c == 32 || CharToNat c - 0x9 <=? 4 || CharToNat c == 0xa0)
+    (IsUnicodeWhitespaceCodePoint (CharToNat c))
+
+-- It checks the whitespace characters (Unicode property White_Space=yes) defined in Unicode 15.0.0 and above the ASCII range.
+type IsUnicodeWhitespaceCodePoint n
+  = n == 0x1680
+  || (0x2000 <=? n && n <=? 0x200A)
+  || n == 0x2028 || n == 0x2029 || n == 0x202F || n == 0x205F || n == 0x3000
+
+type family SymbolLength (length :: Nat) (s :: Maybe (Char, Symbol)) :: Nat where
+  SymbolLength length 'Nothing = length
+  SymbolLength length ('Just '(_, s)) = SymbolLength (length + 1) (UnconsSymbol s)
```

Comment:
Feeling *pretty bad* about a `O(n)` calculation at the type level, especially since many of these are going to be hundreds of characters long. May not be a huge hit. Still, want to be wary...